src/cpp/rtps/writer/StatelessWriter.cpp

```diff
@@ -53,11 +53,7 @@ static bool add_change_to_rtps_group(
         {
             for (uint32_t frag = 1; frag <= n_fragments; frag++)
             {
-                if (group.add_data_frag(*change, frag, inline_qos))
-                {
-                    reader_change->markFragmentsAsSent(frag);
```

Comment:
Why now is not needed to mark fragment as sent?

Reply:
This method is called when there are no flow controllers, and the whole change is sent (i.e. all fragments). As this is the StatelessWriter, if a fragment is lost, the whole change will be lost, so we don't care to mark each fragment as sent, as we are always sending all of them, and the whole change will be marked as sent.